templates/types/streaming/nextjs/app/api/chat/route.ts

```diff
@@ -47,6 +52,10 @@ export async function POST(request: NextRequest) {
         )?.annotations;
     }
 
+    // retrieve document Ids from annotations (if any) and create chat engine with index
+    const ids = retrieveDocumentIds(annotations);
```

Comment:
@leehuwuj FYI: here we're getting the IDs of the user's private documents (via annotations)